### PR TITLE
164796888 Password Reset Functionality

### PR DESCRIPTION
**Pull request by Simplemart17**
#### Description
_As a User_ I want to be able to reset my forgotten password.
An authenticated users can reset their password.
#### Type of change
- [ ] Bug Fix (a non-breaking change which fixes an issue)
- [x] New Feature (a non-breaking change that adds functionality)
- [ ] New Chore (a non-breaking change that adds functionality)
- [ ] Breaking Change (a fix or feature which would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update
#### How Has This Been Tested?
- [x] End to End
- [x] Integration
#### Checklist:
- [x] My code follows the style guidelines of this project
- [x] New and existing unit tests pass locally with my changes
#### Flowchart
flowchart
#### Screenshot
Screenshot 2019-04-18 at 3 41 33 PM

#### PT
[#164796888](https://www.pivotaltracker.com/story/show/164796888)

### CODE REVIEW

**Review by chikeozulumba — COMMENTED** [review]
@Simplemart17 Nice work done. LGTM

File: src/controllers/authentication/passwordReset.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  try {
+    const { resetToken } = req.params;
+    const date = new Date();
+    const user = await User.findOne({ where: { passwordResetToken: resetToken } });
+    if (!user) { return res.status(404).json(errorResponseFormat({ status: 'fail', message: 'Invalid verification token, kindly re-authenticate!' })); }
+    const { resetTokenExpires } = user;
+    if (resetTokenExpires <= date) return res.status(401).json(errorResponseFormat({ status: 'fail', message: 'Token has expired!' }));
+
+    const { password } = req.body;
+    const hashPassword = bcrypt.hashSync(password, bcrypt.genSaltSync(10));
```
Comment (Temmyogunbo) [inline src/controllers/authentication/passwordReset.js]: Let's refactor this and use the method from our user model.

File: src/utils/auth/resetPassword.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import Validator from 'validatorjs';
+/**
+ * @description This is the method for validating password reset field
+ * @param {object} body The request object
+ * @returns {function} Returns validation object
+ */
+export const validateResetField = async (body) => {
```
Comment (Temmyogunbo) [inline src/utils/auth/resetPassword.js]: Why the async?

File: tests/integration/passwordReset.test.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        expect(body.message).to.be.equal('Account associated with this email cannot be found');
+        done();
+      });
+  });
+
+  it('should return error for invalid token', (done) => {
+    agent.put('/api/v1/password-reset/49876jkghdfvghj-dguye89063478-efdvuk789')
+      .send({
+        password: 'tyie56H#JJJ88',
+        password_confirmation: 'tyie56H#JJJ88'
```
Comment (Temmyogunbo) [inline tests/integration/passwordReset.test.js]: Can we put this in a mock file?

**Review by Temmyogunbo — APPROVED** [review]

**Pull request merged**